### PR TITLE
Merging Dashboard to Dev

### PR DESCRIPTION
**Pull request by MohiTheFish**
Ignore the closed PR. It's quite a bit. Will PR more frequently

### CODE REVIEW

File: src/components/Dashboard/cells/sponsors.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+
+
+import Amadeus from '../assets/Sponsors/Amadeus.png';
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/sponsors.js]: let's use lower_snake_case for image assets and folders
path should be ../assets/sponsors/amadeus.png

**Review by brianstrauch — CHANGES_REQUESTED** [review]
Looks good! Before we merge this, let's get rid of all of your commented code (you have a lot of commented CSS), and any other comments that aren't necessary. Also, we'll hook up a linter soon, but if/for blocks should look like `if (condition) {}` with a space after the `if` and before the first curly brace

**Comment by brianstrauch** [comment]
Also we should move the sponsor logos into the `assets` folder in the root directory, since other components (like the Homepage) will use the same images

**Comment by MohiTheFish** [comment]
I changed the names of the sponsors as you suggested. By the way, I stole those images from last year's folder so most of those may have to be thrown out.

File: src/components/Dashboard/index.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    }
+    else{
+      return 'night';
+    }
+  }
+
+  updateTheme(){
+    var newTheme = this.getTheme();
+    //to avoid constantly setting the state of the dashboard, just check if it changed
+    if (newTheme.localeCompare(this.state.theme) !== 0) {
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: I feel like you should just be able to use !== here, right?

File: src/components/Dashboard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      })
+    }
+  }
+  render() {
+    var theme = this.state.theme;
+    return (
+      <div className = {"dashboard-wrapper " + theme}>
+        <div className = "dashboard">
+          <div className = "row top-row">
+            <Time theme={theme}/>
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: I believe there's something in react called contexts where you can set themes easily

File: src/components/Dashboard/index.jsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+  componentDidMount(){
+    this.interval = setInterval(this.updateTheme, 30000); //every 30
+  }
+  componentWillUnmount(){
+    clearInterval(this.interval);
+  }
+
+  getTheme(){
+    var currentTime = new Date();
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: currentTime is kind of verbose, you could call this variable `now` or `time`

File: src/components/Dashboard/index.jsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    var currentHour = currentTime.getHours();
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: Likewise, this could just be `hour` or `h`

File: src/components/Dashboard/index.jsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    var currentMinute = currentTime.getMinutes();
+    currentHour = currentHour + currentMinute/60;
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: Wait so we can have more than 60 minutes in `currentMinute`?

File: src/components/Dashboard/index.jsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+    if (currentHour >= 5 && currentHour < 10) {
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: You could just start with `if (hour < 5) return 'night'`, then all of the other conditionals just have to be `<` inequalities

File: src/components/Dashboard/index.jsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+//All the various cells I'm using have their own js files
+import TwitterFeed from './cells/twitter';
+import CountDown from './cells/countdown';
+import Time from './cells/time';
+import Logo from './cells/logo';
+import Sponsors from './cells/sponsors';
+
+
+class SplitColumn extends React.Component {
```
Comment (brianstrauch) [inline src/components/Dashboard/index.jsx]: I feel like the only components you need are `Dashboard` and `Cell`. Does split column refer to one column of three cells?

File: src/components/Dashboard/cells/time.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    return {
+      hours: hours,
+      minutes: minutes,
+      isAm: isAm
+    }
+  }
+
+  componentDidMount(){
+    this.interval = setInterval(this.setTime, 1000);
+  }
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/time.js]: Separate functions with a space

File: src/components/Dashboard/cells/time.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+    if (hours === 0) {
+      hours = 12;
+    }
+    if (!isAm) {
+      hours -= 12;
+    }
+
+    if (hours < 10) {
+      hours = this.addLeadingZero(hours);
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/time.js]: Maybe write a function called `pad` which adds a zero if the string is too short? Then you don't need to write a new if statement every time you call it

File: src/components/Dashboard/cells/sponsors.js [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+  componentDidMount(){
+    this.interval = setInterval(this.updateImages, this.state.refreshrate);
+  }
+
+  componentWillUnmount(){
+    clearInterval(this.interval);
+  }
+
+
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/sponsors.js]: Get rid of this extra space. Or you could wait until the linter PR goes through, then you can just run the linter on this

File: src/components/Dashboard/cells/sponsors.js [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+      offset: 0,
+      refreshrate: 8 * 1000 //ensure this integer matches the animation duration
+    }
+
+    this.getImage = this.getImage.bind(this);
+    this.updateImages = this.updateImages.bind(this);
+  }
+
+  updateImages(){
+    var newOffset = (this.state.offset+4) % this.state.indices.length;
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/sponsors.js]: Put some space around the `+`

File: src/components/Dashboard/cells/sponsors.js [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+    this.interval = null;
+
+    //list of all our sponsors
+    this.images = [amadeus, bp, c1, caterpillar, citadel, facebook, fulcrum, google,
+    imc, jackson, mirus_research, mozilla, npm, nvisia, optum, particle,
+    rc, schlum, snl, two_sigma, zeit];
+
+    //will contain a list of randomly generated indices
+    var array = new Array(this.images.length);
+    for (var i=0; i<array.length; i++) {
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/sponsors.js]: Put some space around the `=` and `<`

File: src/components/Dashboard/cells/sponsors.js [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+import optum from '../../../assets/sponsors/optum.png';
+import particle from '../../../assets/sponsors/particle.png';
+import rc from '../../../assets/sponsors/rc.png';
+import schlum from '../../../assets/sponsors/schlum.png';
+import snl from '../../../assets/sponsors/snl.png';
+import two_sigma from '../../../assets/sponsors/two_sigma.png';
+import zeit from '../../../assets/sponsors/zeit.svg';
+
+//https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array/2450976#2450976
+function shuffle(array) {
```
Comment (brianstrauch) [inline src/components/Dashboard/cells/sponsors.js]: You could add this to Array.prototype if you wanna be cool

**Review by brianstrauch — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/cells/sponsors.js (hunk 14 above)
Comment (MohiTheFish) [reply src/components/Dashboard/cells/sponsors.js]: The webpack linter said I shouldn't add anything to Array.prototype.
The exact message: 
Array prototype is read only, properties should not be added  no-extend-native

**Pull request closed without merging**